core/cloudflow-operator-actions/src/main/scala/cloudflow/operator/action/Action.scala

```diff
@@ -31,6 +31,12 @@ import skuber.api.patch.Patch
  */
 sealed trait Action {
 
+  /**
+   * Executes the action using a KubernetesClient.
+   * Returns the created or modified resource, or None if the resource is deleted.
+   */
+  def execute(client: KubernetesClient)(implicit sys: ActorSystem, ec: ExecutionContext, lc: LoggingContext): Future[Action]
```

Comment:
not related to this PR, but, in general it would be nice to move:
```scala
(implicit sys: ActorSystem, ec: ExecutionContext, lc: LoggingContext)
```
at the constructor level instead of the methods